### PR TITLE
Closes #5675 improve wpx hosting compatibility

### LINKED ISSUE #5675: Improve WPX hosting compatibility
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
For WPX hosting we're removing the expires rules for HTML. That's not the best we could do, as it introduces some side effects (HTML content not refreshed)

**To Reproduce**
1. Go to WPX website
2. Check the headers for HTML request
3. It's set to 1 month

**Expected behavior**
The best we can do for compatibility is to make sure that HTML document will have the following headers:
- public
- max-age=0
- s-maxage=3600
- stale-while-revalidate=21600

**Additional context**
At this point we can easily remove the part that's deleting our expires rules:
https://github.com/wp-media/wp-rocket/blob/5ca5716ea7452077c8c4d784619bb8950d4019da/inc/ThirdParty/Hostings/WPXCloud.php#L48-L53
The browsers will use the correct max-age header again.

We also need to add the following Cache-Control headers:
- s-maxage=3600
- stale-while-revalidate=21600
So their CDN can take advantage of it. The values were consulted with Aleksandar, their Systems Administrators Team Lead.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description
Improve wpx hosting compatibility

Fixes #5675 

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

Yes, We need to append the cache control header for only mime type text/html content and previous grooming would only work with a matching html file.

## How Has This Been Tested?

Automatic & Manual Tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -184,7 +184,8 @@
 		    "@test-integration-rank-math-seo",
 		    "@test-integration-all-in-seo-pack",
 		    "@test-integration-seopress",
-		    "@test-integration-the-events-calendar"
+		    "@test-integration-the-events-calendar",
+			"@test-integration-wpxcloud"
```
Comment (CrochetFeve0251) [inline composer.json]: @jeawhanlee is it possible to correct indentation?

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

**Review by CrochetFeve0251 — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@jeawhanlee thanks for the PR, Partial Cache clear is not working there, can you please check? (was working fine on trunk 3.12.5.3)
steps:
1- Cache a page incognito (Cache HIT) 
2- Edit that page
3- Refresh the page incognito => text is not updated

As per @vmanthos 

> It's standard Varnish functionality and that should be working. :thinking_face:
There was an issue with clearing the homepage and partial cache clearing:
https://github.com/wp-media/wp-rocket/pull/5515#issuecomment-1298202478
But supposedly WPX resolved that. Maybe it's the same.

**Comment by vmanthos** [comment]
I've logged the purge request to WPX's Varnish:
```PHP
URL:https://wpxnew.rocketlabsqa.ovh/hello-world/?regex
Purge URL:http://127.0.0.1:6081/hello-world/.*
Array
(
    [method] => PURGE
    [blocking] => 
    [redirection] => 0
    [headers] => Array
        (
            [host] => wpxnew.rocketlabsqa.ovh
            [X-Purge-Method] => regex
        )

)
```

Nothing has changed on our side compared to the [previous time](https://github.com/wp-media/wp-rocket/pull/5515#issuecomment-1298112131) I did the same.

@alxtv Can you possibly look into this and check if it's something on your side? :pray:

**Comment by alxtv** [comment]
According to our logs, it's working fine.

I will double check it.

PURGE

**Comment by jeawhanlee** [comment]
Thanks @piotrbak, we found out that partial cache clearing will only happen when permalink has a trailing slash as we are only matching urls with a trailing slash [here](https://github.com/wp-media/wp-rocket/blob/60b76bc8bc9a366bf5b9644894df8203bba103fd/inc/Addon/Varnish/Subscriber.php#L84) and [here](https://github.com/wp-media/wp-rocket/blob/60b76bc8bc9a366bf5b9644894df8203bba103fd/inc/Addon/Varnish/Subscriber.php#L98)
@Mai could you please confirm this?

**Comment by piotrbak** [comment]
@Mai-Saad @jeawhanlee It should be the same on the trunk though.

@alxtv 👋 It looks like you're caching both:
1. https://wpxnew.rocketlabsqa.ovh/hello-world/
2. https://wpxnew.rocketlabsqa.ovh/hello-world

But when we send the PURGE request like this one:
- http://127.0.0.1:6081/hello-world/.*

It only clears the 1st URL. I've just tried sending the PURGE requests without trailing slashes:
- http://127.0.0.1:6081/hello-world.*
- http://127.0.0.1:6081/hello-world(.*)

But it didn't work as expected.

**Comment by alxtv** [comment]
We can fix that and handle http://127.0.0.1:6081/hello-world.* or http://127.0.0.1:6081/hello-world(.*) PURGE like requests.
Just let me know how it will be better for you.

Meanwhile I've tested the new "cache-control" header and it works as expected -> no browser cache for text/html, but cached on the CDN.

**Comment by piotrbak** [comment]
@alxtv Great about the browser caching! 🙌 

From our perspective it would be the best if `http://127.0.0.1:6081/hello-world/.*` would clear both:
-  http://127.0.0.1:6081/hello-world/
-  http://127.0.0.1:6081/hello-world
(in that case, our PR is ready-to-go)

Most likely it should not be possible to have both of them accessible though. If it's not doable, then sending that kind of request ` http://127.0.0.1:6081/hello-world.*` would be the best. We'd need to make sure it also clears both URLs.

**Comment by alxtv** [comment]
@piotrbak we will clear both http://127.0.0.1:6081/hello-world and http://127.0.0.1:6081/hello-world/ if we receive PURGE request for http://127.0.0.1:6081/hello-world/.*
Will confirm once it's deployed on the whole CDN.

**Comment by piotrbak** [comment]
@alxtv That's perfect, thank you 🙏 

@Mai-Saad Partial cache clearing case will be handled on the WPX side.

**Review by Mai-Saad — APPROVED** [review]
Working as expected in terms of adding headers and partial cache clear while permalinks with `/`.
notes:
- after deactivating WPR, headers can still exist for a while in the browser (probably due to browser cache, however, WPR rules were removed from htaccess)
- Partial cache clearing while permalinks without `/` will be handled on WPX's side
[testrail-report-449.pdf](https://github.com/wp-media/wp-rocket/files/10722077/testrail-report-449.pdf)

**Comment by alxtv** [comment]
> @piotrbak we will clear both http://127.0.0.1:6081/hello-world and http://127.0.0.1:6081/hello-world/ if we receive PURGE request for http://127.0.0.1:6081/hello-world/.* Will confirm once it's deployed on the whole CDN.

This is done.

**Pull request merged**